File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
    let low = 0
    let high = numbers.length -1

    while(low < high) {
        let sum = numbers[high] + numbers[low]
        if(sum === target) {
            return [low + 1, high + 1]
        } else if(sum > target) {
            high--
        } else {
            low++
        }
    }

    return []
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo, hi = 0, len(numbers) - 1

        while lo <= hi:
            m = (hi + lo) // 2
            ans = numbers[lo] + numbers[hi]
            print(lo, hi, ans, target > ans)

            if ans == target:
                return [lo + 1, hi + 1]
            
            if target < ans:
                hi -= 1
            else:
                lo += 1
        

            